server/src/main/java/io/druid/segment/realtime/appenderator/AppenderatorImpl.java

```diff
@@ -1119,15 +1096,16 @@ public String apply(SegmentIdentifier input)
   )
   {
     // Ensure no future writes will be made to this sink.
-    sink.finishWriting();
+    if (sink.finishWriting()) {
+      // Decrement this sink's rows from rowsCurrentlyInMemory (we only count active sinks).
+      rowsCurrentlyInMemory.addAndGet(-sink.getNumRowsInMemory());
```

Comment:
Hmm, is this valid? It looks that this would not be called if [this](https://github.com/apache/incubator-druid/pull/6129/files#diff-940e5274e2cfb880c70e6249ea864f63R605) is executed first.